lib/Board.rb

```diff
@@ -0,0 +1,34 @@
+ require_relative "UI"
+
+class Board
+    attr_accessor :board 
+
+    include UserInterface
+    def initialize()
+        @board = [0,1,2,3,4,5,6,7,8]
+    end 
+
+    def add_piece(position, piece)
+        if is_position_available(position)
+            @board[position] = piece
+            true
+        else 
+            self.print_taken
+            false
+        end
+    end
+
+    def is_position_available(pos) 
```

Comment:
Do we need to expose this method publicly? In other words, are other objects calling this method? If not then it should just be under a `private` keyword